Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -87,7 +88,22 @@ static String escape_regexp_pattern(const RegExpObject& regexp_object)
     return pattern;
 }
 
-static void increment_last_index(GlobalObject& global_object, Object& regexp_object)
+// 22.2.5.2.3 AdvanceStringIndex ( S, index, unicode ), https://tc39.es/ecma262/#sec-advancestringindex
+size_t advance_string_index(String const& string, size_t index, bool unicode)
+{
+    if (!unicode)
+        return index + 1;
+
+    Utf8View view(string);
+
+    if (index + 1 >= view.length())
+        return index + 1;
+
+    auto it = view.iterator_at_byte_offset(index);
```

Comment:
as the name suggests this expects a byte offset, not a codepoint index, so this is not correct behaviour.